crates/matrix-sdk/src/client/builder.rs

```diff
@@ -1061,6 +1072,34 @@ pub(crate) mod tests {
         assert_eq!(client.sliding_sync_proxy(), Some("https://localhost:9012".parse().unwrap()));
     }
 
+    #[async_test]
+    #[cfg(feature = "experimental-sliding-sync")]
+    async fn test_discovery_well_known_with_simplified_sliding_sync() {
+        // Given a base server with a well-known file that points to a homeserver with a
+        // sliding sync proxy.
+        let server = MockServer::start().await;
+        let homeserver = make_mock_homeserver().await;
+        let mut builder = make_ffi_builder();
+
+        Mock::given(method("GET"))
+            .and(path("/.well-known/matrix/client"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(make_well_known_json(
+                &homeserver.uri(),
+                Some("https://localhost:1234"),
+            )))
+            .mount(&server)
+            .await;
+
+        // When building a client for simplified sliding sync with the base server.
+        builder = builder.simplified_sliding_sync(true);
+        builder = builder.server_name_or_homeserver_url(server.uri());
+        let _client = builder.build().await.unwrap();
+
+        // Then a client should not use the discovered sliding sync proxy.
+        #[cfg(feature = "experimental-sliding-sync")]
```

Comment:
nit: don't need this cfg because the test is guarded against the same cfg guard

Then we could rename `_client` to `client` since it'd be always used.